src/ClientOptionsTrait.php

```diff
@@ -317,4 +351,26 @@ private function isBackwardsCompatibilityMode(): bool
     {
         return false;
     }
+
+    /**
+     * @param null|false|LoggerInterface $logger
+     * @param string $options
+     */
+    private function logConfiguration(null|false|LoggerInterface $logger, string $options): void
+    {
+        if (!$logger) {
+            return;
+        }
+
+        $configurationLog = [
+            'timestamp' => date(DATE_RFC3339),
+            'severity' => strtoupper(LogLevel::DEBUG),
+            'processId' => getmypid(),
+            'jsonPayload' => [
+                'suppliedConfiguration' => $options
```

Comment:
Can we make this something else? Such as `clientConfiguration`? as it is now, we don't know what configuration is being supplied. If you want to be super explicit, we could make it `userSuppliedClientConfiguration`.

Another issue is that there's no indication of what client this is. Would it make sense to have the classname or API endpoint in there as well? Something like:

```php
'jsonPayload' => [
    'clientName' => $this->serviceName, // could also be "get_class($this)"
    'clientConfiguration' => $options,
]
```